realm/realm-library/src/testUtils/java/io/realm/TestHelper.java

```diff
@@ -1249,4 +1252,39 @@ public static void populateLinkedDataSet(Realm realm) {
         realm.commitTransaction();
     }
 
+    /**
+     * This method will kill all tasks then shutdown and replace the SyncManager.NETWORK_POOL_EXECUTOR
+     * with a fresh and empty instance. This should only be called when exiting tests.
+     *
+     * If the build does not support Sync, this method will do nothing
+     */
+    public static void waitForNetworkThreadExecutorToFinish() {
+        try {
+            Class syncManager = Class.forName("io.realm.SyncManager");
+            Field field = syncManager.getDeclaredField("NETWORK_POOL_EXECUTOR");
+            ThreadPoolExecutor pool = (ThreadPoolExecutor) field.get(null);
```

Comment:
minor: for performance reason, you can make the reflection code static for caching, since this is called more often for integration tests